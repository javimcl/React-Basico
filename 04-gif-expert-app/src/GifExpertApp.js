import React, { useState } from 'react'
import { AddCategory } from './componentes/AddCategory';
import { GifGrid } from './componentes/GifGrid';

export const GifExpertApp = () => {

   // const categorias = ['One Puch', 'Samurai X', 'Dragon Ball'];

   //const [categories, setCategories] = useState(['One Puch', 'Samurai X', 'Dragon Ball']);

   const [categories, setCategories] = useState(['One Puch']);


// const handleAdd = () => {
//     //setCategories(...categories, 'HunterXHunter' );
//     setCategories(cats => [...cats, 'HunterXHunter'] ); 
// }

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories = { setCategories }/>
            <hr/>
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <ol>
                {
                    categories.map(category => (
                        <GifGrid key={category} category={category}/>
                        //  <li key={categoria}>{categoria}</li>
                    ))
                }
            </ol>
        </>
    )
}
