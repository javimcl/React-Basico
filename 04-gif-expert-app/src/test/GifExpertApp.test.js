import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp'

describe('Prueba de <GifExpertApp/>', () => {
    

    test('Prueba de correctamente ', () => {
        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();

    })

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['OnePunch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    
    
})
