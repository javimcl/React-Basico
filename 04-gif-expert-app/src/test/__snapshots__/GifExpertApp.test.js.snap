// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <GifExpertApp/> Prueba de correctamente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Prueba de <GifExpertApp/> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="OnePunch"
      key="OnePunch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
